pkg/tarutil/tarutil.go

```diff
@@ -10,78 +10,52 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package main
+package tarutil
 
 import (
 	"archive/tar"
 	"bytes"
 	"compress/gzip"
+	"fmt"
 	"io"
-	"io/ioutil"
 	"net/http"
 	"net/url"
 	"path"
 	"strings"
 
-	"github.com/ghodss/yaml"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-	helmrepo "k8s.io/helm/pkg/repo"
-	log "k8s.io/klog/v2"
+	chart "github.com/kubeapps/kubeapps/pkg/chart/models"
+	httpclient "github.com/kubeapps/kubeapps/pkg/http-client"
 )
 
-const (
-	readme = "readme"
-)
-
-//
-// TODO some of this functionality already exists in asset-syncer but is private
-// so it needs to be re-packaged so that it can be re-used
-//
-func getHelmIndexFileFromURL(indexURL string) (*helmrepo.IndexFile, error) {
-	log.Infof("+getHelmIndexFileFromURL(%s) 1", indexURL)
-	// Get the response bytes from the url
-	response, err := http.Get(indexURL)
+func FetchDetailFromTarball(name string, chartTarballURL string, userAgent string, authz string, netClient httpclient.Client) (map[string]string, error) {
```

Comment:
I'd rename this fn to sth like `FetchChartDetailFromTarball` (to highlight it's just for chart tarballs, not any tarball) since it is now publicly available under the `tarutil` package

Reply:
Idem here about documenting

Reply:
will do. Thanks